assembly/src/parsers/ast/mod.rs

```diff
@@ -0,0 +1,563 @@
+use crate::MODULE_PATH_DELIM;
+
+use super::{AssemblyError, Token, TokenStream, Vec};
+
+use std::collections::BTreeMap;
+
+mod types;
+use types::{Instruction, Node, Program};
+
+mod io_ops;
+mod stack_ops;
+mod u32_ops;
+
+#[cfg(test)]
+pub mod tests;
+
+// AST Parser that holds internal state to generate correct ASTs.
+#[allow(dead_code)]
+struct Parser {
+    uses: BTreeMap<String, String>,
```

Comment:
Does this assume that we need to instantiate a new parser to parse each program? If so, I'm not sure if that's the best approach. If we need to pass around some common info during parsing process, maybe it would make sense to create a context struct (e.g., `ParserContext`) which would contain information about imports, procedures etc.

Also nit: I'd probably name this `imports` rather than `uses`.

Reply:
Got it, that works too!